pkg/etcdcli/cached_members_test.go

```diff
@@ -0,0 +1,364 @@
+package etcdcli
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	u "github.com/openshift/cluster-etcd-operator/pkg/testutils"
+	"github.com/stretchr/testify/require"
+
+	"go.etcd.io/etcd/api/v3/etcdserverpb"
+)
+
+func TestCachedMembers_refresh(t *testing.T) {
+	testCases := []struct {
+		name            string
+		etcdMembersList []*etcdserverpb.Member
+		expEtcdMembers  []*etcdserverpb.Member
+	}{
+		{
+			"all healthy members",
```

Comment:
one last nit-pick: if it's just a test, you can directly write it down without the t.Run indirection

Reply:
But is this 'refreshTest' sufficient ?

I used `time.Sleep()` 

What would be better option ?

Reply:
should be fine, another option would be to have some kind of callback that you could register on refresh and assert that one is called

Reply:
Like ?

`mock or stub`

I would like to add this actually -   i used to do it java but never done it in go

Reply:
i think it is too complex for a simple test .. lets keep it this way 👍🏽

Reply:
sounds good!